search-processors/src/test/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessorTests.java

```diff
@@ -220,4 +225,47 @@ public void testProcessResponseMissingContextField() throws Exception {
 
         assertTrue(exceptionThrown);
     }
+
+    public void testProcessorFactoryFeatureDisabled() throws Exception {
+        Client client = mock(Client.class);
+        Map<String, Object> config = new HashMap<>();
+        config.put(GenerativeQAProcessorConstants.CONFIG_NAME_MODEL_ID, "xyz");
+        config.put(GenerativeQAProcessorConstants.CONFIG_NAME_CONTEXT_FIELD_LIST, List.of("text"));
+
+        Processor.Factory factory = new GenerativeQAResponseProcessor.Factory(client, () -> false);
+
+        try {
+            factory.create(null, "tag", "desc", true, config, null);
+        } catch (MLException e) {
+            assertEquals(GenerativeQAProcessorConstants.FEATURE_NOT_ENABLED_ERROR_MSG, e.getMessage());
+        }
+    }
+
+    // Use this only for the following test case.
+    private boolean featureEnabled001;
```

Comment:
you think you can use @Before function to reset the variable and then before start the test in the test you can set your variable to the desired value.

Reply:
I need to be able to change this boolean multiple times during a single test case run.

Reply:
I don't think resetting using _before_ is needed?